reflex/components/sonner/toast.py

```diff
@@ -27,6 +30,29 @@
 toast_ref = Var.create_safe("refs['__toast']")
 
 
+class ToastAction(Base):
+    """A toast action that render a button in the toast."""
+
+    label: str
+    on_click: Any
+
+
+@serializer
+def serialize_action(action: ToastAction) -> dict:
+    """Serialize a toast action.
+
+    Args:
+        action: The toast action to serialize.
+
+    Returns:
+        The serialized toast action with on_click formatted to queue the given event.
+    """
+    return {
+        "label": action.label,
+        "onClick": format.format_queue_events(action.on_click),
+    }
+
+
 class PropsBase(Base):
```

Comment:
I think `PropsBase` should be exposed under `_x` too.
I've been playing with some libs that make extensive use of hooks, and the feature from this would be needed for them too.

Maybe in its own file since the code for it is not specific to sonner/toast